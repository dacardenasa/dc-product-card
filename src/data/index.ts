import { noImageURL } from "../constants";

export const product1 = {
  id: '1',
  title: 'Cofee Mug - sin imagen',
  // uriImage: "/coffee-mug.png"
};

export const product2 = {
  id: '1',
  title: 'Cofee Mug - CardById',
  uriImage: noImageURL,
};
