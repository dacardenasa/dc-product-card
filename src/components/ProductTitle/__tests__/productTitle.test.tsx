import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle } from '..';
import { ProductCard } from '../../ProductCard';
import { product1 } from '../../../data';

describe('ProductTitle', () => {
  test('render correctlly components', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Cutom Product" className="custom-class" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('render component without title prop correctlly', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
