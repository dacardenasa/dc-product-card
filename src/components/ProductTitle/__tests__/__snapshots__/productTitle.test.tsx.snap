// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle render component without title prop correctlly 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Cofee Mug - sin imagen
  </span>
</div>
`;

exports[`ProductTitle render correctlly components 1`] = `
<span
  className="productDescription custom-class"
>
  Cutom Product
</span>
`;
